frontend/src/main/components/RecommendationRequest/RecommendationRequestForm.js

```diff
@@ -0,0 +1,160 @@
+import { Button, Form, Row, Col } from "react-bootstrap";
+import { useForm } from "react-hook-form";
+import { useNavigate } from "react-router-dom";
+
+function RecommendationRequestForm({
+  initialContents,
+  submitAction,
+  buttonLabel = "Create",
+}) {
+  // Stryker disable all
+  const {
+    register,
+    formState: { errors },
+    handleSubmit,
+  } = useForm({ defaultValues: initialContents || {} });
+  // Stryker restore all
+
+  const navigate = useNavigate();
+
+  // For explanation, see: https://stackoverflow.com/questions/3143070/javascript-regex-iso-datetime
+  // Note that even this complex regex may still need some tweaks
+
+  // Stryker disable Regex
+
+  // Stryker disable next-line all
+
+  return (
+    <Form onSubmit={handleSubmit(submitAction)}>
+      <Row>
+        {initialContents && (
+          <Col>
+            <Form.Group className="mb-3">
+              <Form.Label htmlFor="id">Id</Form.Label>
+              <Form.Control
+                data-testid="RecommendationRequestForm-id"
+                id="id"
+                type="text"
+                {...register("id")}
+                value={initialContents.id}
+                disabled
+              />
+            </Form.Group>
+          </Col>
+        )}
+      </Row>
+
+      <Row>
+        <Col>
+          <Form.Group className="mb-3">
+            <Form.Label htmlFor="professorName">Professor Name</Form.Label>
+            <Form.Control
+              data-testid="RecommendationRequestForm-professorName"
+              id="professorName"
+              type="text"
+              isInvalid={Boolean(errors.professorName)}
+              {...register("professorName", {
+                required: "professorName is required.",
+              })}
+            />
+            <Form.Control.Feedback type="invalid">
+              {errors.professorName?.message}
+            </Form.Control.Feedback>
+          </Form.Group>
+        </Col>
```

Comment:
What is this field for?  Note that the POST method only takes professorEmail..

Keep in mind that the purpose of the form is to gather information that is passed to the POST endpoint to create a new RecommendationRequest.

If there isn't a parameter at the POST endpoint for the field, then why ask the user for the data? It just gets thrown away...

```
    public RecommendationRequest postRecommendationRequest(
        @Parameter(name="professorEmail") @RequestParam String professorEmail,
        @Parameter(name="recommendationType") @RequestParam String recommendationType,
        @Parameter(name="details") @RequestParam String details)
        throws JsonProcessingException {
```